Fix pages overflow

mateusabelli opened the pull request:
## Summary

This PR is a continuation of #14 keeping the code element inline and also applying the change reviewed by @jessarcher to make the CSS selectors more explicit.

I've tried all pages to look for more overflow issues and I found at least two pages that the tables were also overflowing, I applied similar tailwind properties to solve it as well.

## Demonstration

### :x: Before the changes

![before_bladeediting-chirps](https://github.com/laravel/bootcamp.laravel.com/assets/43862225/2645779f-397e-4995-a3e0-f2bf649125c4)


![before-bladecreating-chirps](https://github.com/laravel/bootcamp.laravel.com/assets/43862225/a43a552b-93de-4acd-be39-74858e37c347)

### ✔ After the changes

![bladeediting-chirps](https://github.com/laravel/bootcamp.laravel.com/assets/43862225/65ed8289-b43b-4838-b77b-eabbc0ff795c)

![bladecreating-chirps](https://github.com/laravel/bootcamp.laravel.com/assets/43862225/54d510bf-9b9c-4d50-b6dc-6cc0186b90ab)

## Breakdown

```css
li code,
p code {
    @apply inline-flex max-w-full overflow-scroll
}

table {
    @apply block max-w-full overflow-scroll
}
```

`inline-flex` or `display: inline-flex` is similar to `display: inline-block` but using the `flex` property. Unfortunately I didn't understood quite well so here is the [link](https://stackoverflow.com/questions/27418104/whats-the-difference-between-displayinline-flex-and-displayflex) for more info about it. I had to use it instead of `inline-block` because it was displaying a slightly different positioning on the code elements.

`max-w-full` or `max-width: 100%` is going to allow the code element to occupy all of its parent width, making it possible for the next attribute to control the overflow.

`overflow-scroll` or `overflow: scroll` will detect if the content is trying to go beyond its parent's width and add scroll bars to it.

jessarcher commented on resources/css/_code.css:
```diff
@@ -15,6 +15,15 @@ pre code.torchlight .summary-caret {
+    @apply inline-flex max-w-full overflow-scroll
+}
+
+table {
+    @apply block max-w-full overflow-scroll
+}
+
```
This is great! I think it should be moved to `app.css` as this file is specific to code snippets.

mateusabelli replied:
That's cool! I've moved it to `app.css` but I did change `overflow-scroll` to `overflow-auto` I'll be leaving a comment comparing the two, with the reason why I did. I'm always open for feedback!

jessarcher commented on resources/css/_code.css:
```diff
@@ -15,6 +15,15 @@ pre code.torchlight .summary-caret {
     @apply mr-1;
 }
 
+li code,
+p code {
+    @apply inline-flex max-w-full overflow-scroll
+}
+
```
The gap beneath these looks pretty weird, and I'm unsure where it's coming from. I'm also not a fan of how it creates a full-width rectangle:

![image](https://github.com/laravel/bootcamp.laravel.com/assets/4977161/4ed5b115-45be-4fab-a0a2-0173860ee59d)

I think I'd prefer to just change `overflow-wrap` so that it will break mid-word when necessary:

```suggestion
li code,
p code {
    @apply break-words;
}
```

![image](https://github.com/laravel/bootcamp.laravel.com/assets/4977161/cf34a98f-6687-4b7f-89b1-e29b09879f0d)

mateusabelli replied:
I agree with you, I did the changes as requested in my latest push. Please let me know what do you think.

jessarcher reviewed (changes requested):
Thanks!

mateusabelli commented:
Hello @jessarcher 

Thanks for reviewing my changes. I did some extra modification on the table code, instead of `overflow-scroll` I changed it to `overflow-auto` so that the scrollbars only appear when they are actually needed. 

When I did my tests and demos with GIFs the User Agent on responsive mode was set to "Mobile" changing it to "Desktop" made me realize that the scrollbars were visible at all times. 

I also added a descriptive comment to follow along the previous code on `app.css`. Please let me know what do you think

<details>
<summary>Small suggestion</summary>

While doing some tests I noticed something else, it's a minor style thing but I think we could enhance the experience by having dark scrollbars with the dark theme, it doesn't take much code:
```css
:root {
  color-scheme: dark
}
```

I'm not really sure if you would like it or where to put it, so I'm just going to leave it here as a suggestion as well. Heres a quick demo of it:

<table>
<td><img src="https://github.com/laravel/bootcamp.laravel.com/assets/43862225/f2720656-2c52-41ea-8726-df7a032f9ff6" /></td>
<td><img src="https://github.com/laravel/bootcamp.laravel.com/assets/43862225/d867558b-e581-49db-bd05-695e45456237" /></td>
</table>

</details>

jessarcher commented:
Thanks, @mateusabelli!

Regarding the `color-scheme` suggestion, I'm a bit confused about the property's intended purpose, even after reading https://developer.mozilla.org/en-US/docs/Web/CSS/color-scheme. Given we support dark and light modes, I'd have thought it should be `color-scheme: light dark;`. On Firefox+Linux it doesn't seem to have any effect so I'm hesitant to change it without being able to see what it's doing.
